packages/reactivity/src/effect.ts

```diff
@@ -122,62 +101,33 @@ export class ReactiveEffect<T = any> {
     if (!this.active) {
       return this.fn()
     }
-    let parent: ReactiveEffect | undefined = activeEffect
     let lastShouldTrack = shouldTrack
-    while (parent) {
-      if (parent === this) {
-        return
-      }
-      parent = parent.parent
-    }
+    let lastEffect = activeEffect
     try {
-      this.parent = activeEffect
-      activeEffect = this
       shouldTrack = true
-
-      trackOpBit = 1 << ++effectTrackDepth
-
+      activeEffect = this
+      this._runnings++
       cleanupEffect(this)
       return this.fn()
     } finally {
-      if (effectTrackDepth <= maxMarkerBits) {
-        finalizeDepMarkers(this)
-      }
-
-      trackOpBit = 1 << --effectTrackDepth
-
-      activeEffect = this.parent
+      this._runnings--
+      activeEffect = lastEffect
       shouldTrack = lastShouldTrack
-      this.parent = undefined
-
-      if (this.deferStop) {
-        this.stop()
-      }
     }
   }
 
   stop() {
-    // stopped while running itself - defer the cleanup
-    if (activeEffect === this) {
-      this.deferStop = true
-    } else if (this.active) {
+    if (this.active) {
       cleanupEffect(this)
-      if (this.onStop) {
-        this.onStop()
-      }
+      this.onStop?.()
       this.active = false
     }
   }
 }
 
 function cleanupEffect(effect: ReactiveEffect) {
-  const { deps } = effect
-  if (deps.length) {
-    for (let i = 0; i < deps.length; i++) {
-      deps[i].delete(effect)
-    }
-    deps.length = 0
-  }
+  effect._trackId++
```

Comment:
After cleaning up an effect (scope), this means that `deps` arrays will still point towards it. In contrast to the previous code which actually removed the reference.
Potentially this could cause a memory leak.

I think you revert to the old solution (though only in the `stop` function).